scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/util/DateTimeComparator.java

```diff
@@ -0,0 +1,31 @@
+package org.cloudfoundry.autoscaler.scheduler.util;
+
+import java.util.Comparator;
+
+public class DateTimeComparator implements Comparator<SpecificDateScheduleDateTime> {
+
+	public int compare(SpecificDateScheduleDateTime first, SpecificDateScheduleDateTime second) {
+		if ((first == null) && (second == null))
+			return 0;
+		if (first == null)
+			throw new RuntimeException("first object is null in DateTimeComparator");
+		if (second == null)
+			throw new RuntimeException("second object is null in DateTimeComparator");
+
+		Long firstValue = first.getStartDateTime();
+		Long secondValue = second.getStartDateTime();
+
+		if (firstValue == null)
+			throw new RuntimeException("firstValue is null in DateTimeComparator");
+		if (secondValue == null)
+			throw new RuntimeException("secondValue is null in DateTimeComparator");
+
+		if (firstValue > secondValue) {
```

Comment:
read the docs for Comparator.

return firstValue - secondValue is all you need.